src/vs/base/browser/ui/iconLabel/iconLabelHover.ts

```diff
@@ -35,18 +35,69 @@ export interface ICustomHover extends IDisposable {
 	 * Allows to programmatically hide the hover.
 	 */
 	hide(): void;
+
+	/**
+	 * Updates the contents of the hover.
+	 */
+	update(tooltip: string | IIconLabelMarkdownString | HTMLElement): void;
 }
 
+type HoverWidgetContent = IMarkdownString | string | HTMLElement | undefined;
+class UpdatableHoverWidget {
```

Comment:
Should this maybe `implements IDisposable` to make clear it needs to be disposed?